rustls/src/crypto/aws_lc_rs/hpke.rs

```diff
@@ -418,23 +420,42 @@ impl Hpke for &Suite {
     }
 }
 
-impl Debug for Suite {
+impl<const KEY_SIZE: usize, const KDF_SIZE: usize> Debug for Suite<KEY_SIZE, KDF_SIZE> {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
         self.suite.fmt(f)
     }
 }
 
+/// A [Hpke] suite implementation backed by `aws-lc-rs` cryptography.
+pub trait AwsLcRsHpkeSuite: Hpke {
```

Comment:
Not sure I fully understand why this intermediate trait is needed?

Reply:
I tried to explain in the commit message but might have missed the mark. We can't collect up the different const-generic parameterized impls into one slice without a trait abstracting over them. The trait is also necessary to be able to match a suite ID to a `dyn Hpke` since that's not possible with just that interface. Lastly we need a way to convert from the `AwsLcRsHpkeSuite` to a `Box<dyn Hpke>` without using unstable trait upcasting.